garak/resources/loaders/data_sources.py

```diff
@@ -0,0 +1,195 @@
+        else:
+            raise ValueError(
+                f"Unsupported file format: {suffix}. Supported: .txt, .json"
+            )
+    
+    @staticmethod
+    def _load_txt(file_path: Path) -> List[str]:
+        """Load from text file (one item per line)."""
+        with open(file_path, 'r', encoding='utf-8') as f:
+            items = [line.strip() for line in f if line.strip()]
+        
+        if not items:
+            raise ValueError(f"No data found in {file_path}")
+        
+        return items
+    
+    @staticmethod
+    def _load_json(file_path: Path, **kwargs) -> List[str]:
+        """Load from JSON file.
+        
+        Supports:
+        - Array: ["item1", "item2"]
+        - Object: {"prompts": ["item1", "item2"], "metadata": {...}}
+        
+        If metadata_callback is provided, calls it with metadata dict.
+        """
+        with open(file_path, 'r', encoding='utf-8') as f:
+            try:
+                data = json.load(f)
+            except json.JSONDecodeError as e:
+                raise ValueError(f"Invalid JSON in {file_path}: {e}")
+        
+        # Handle array format
+        if isinstance(data, list):
+            items = [str(item).strip() for item in data if item]
+        # Handle object format
+        elif isinstance(data, dict):
+            # Extract items from 'prompts' or first list-valued key
+            if 'prompts' in data:
+                items = [str(p).strip() for p in data['prompts'] if p]
+            else:
+                # Try to find first list value
+                for key, value in data.items():
+                    if isinstance(value, list):
+                        items = [str(v).strip() for v in value if v]
+                        break
+                else:
+                    raise ValueError(
+                        f"No list found in JSON. Expected array or object with 'prompts' key"
+                    )
+            
+            # Pass metadata to callback if provided
+            metadata_callback = kwargs.get('metadata_callback')
+            if metadata_callback and callable(metadata_callback):
+                metadata = {k: v for k, v in data.items() if k != 'prompts'}
+                metadata_callback(metadata)
+        else:
+            raise ValueError(f"JSON must be array or object")
+        
+        if not items:
+            raise ValueError(f"No data found in {file_path}")
+        
+        return items
+
+
+class HTTPDataLoader:
+    """Loader for HTTP(S) URLs."""
+    
+    @staticmethod
+    def load(url: str, **kwargs) -> List[str]:
+        """Load data from HTTP(S) URL.
+        
+        Detects format (.txt or .json) and parses accordingly.
+        """
+        try:
+            import requests
+        except ImportError:
+            raise ValueError(
+                "requests library required for HTTP loading. "
+                "Install with: pip install requests"
+            )
```

Comment:
should we put this in `pyproject.toml`? (and ditch the handling here). thought `requests` was a core lib - perhaps not!